src/main/java/com/cozyhome/onlineshop/orderservice/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,75 @@
+package com.cozyhome.onlineshop.orderservice.service.impl;
+
+import com.cozyhome.onlineshop.dto.order.OrderDto;
+import com.cozyhome.onlineshop.dto.order.OrderNumberDto;
+import com.cozyhome.onlineshop.orderservice.model.Delivery;
+import com.cozyhome.onlineshop.orderservice.model.Order;
+import com.cozyhome.onlineshop.orderservice.model.OrderItem;
+import com.cozyhome.onlineshop.orderservice.model.OrderNumber;
+import com.cozyhome.onlineshop.orderservice.model.enums.OrderStatus;
+import com.cozyhome.onlineshop.orderservice.repository.DeliveryRepository;
+import com.cozyhome.onlineshop.orderservice.repository.OrderNumberRepository;
+import com.cozyhome.onlineshop.orderservice.repository.OrderRepository;
+import com.cozyhome.onlineshop.orderservice.service.OrderService;
+import com.cozyhome.onlineshop.orderservice.service.builder.DeliveryBuilder;
+import com.cozyhome.onlineshop.orderservice.service.builder.OrderItemBuilder;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderNumberRepository orderNumberRepository;
+    private final OrderItemBuilder orderItemBuilder;
+    private final DeliveryBuilder deliveryBuilder;
+
+
+    @Override
+    public OrderNumberDto newOrder(OrderDto orderDto, String userId) {
+
+        List<OrderItem> orderItems = orderItemBuilder.buildOrderItems(orderDto.getOrderItems());
+
+        Order order = Order.builder()
+                .firstName(orderDto.getFirstName())
+                .lastName(orderDto.getLastName())
+                .phoneNumber(orderDto.getPhoneNumber())
+                .orderItem(orderItems)
+                .orderStatus(OrderStatus.NEW)
+                .createdAt(LocalDateTime.now())
+                .modifiedAt(LocalDateTime.now())
+                .build();
+
+        if (orderDto.getDelivery().getStreet() == null) {
```

Comment:
more obvious to check if (deliveryCompanyName != null)

Reply:
done